pkg/agent/run.go

```diff
@@ -97,6 +97,11 @@ func run(ctx context.Context, cfg cmds.Agent, proxy proxy.Proxy) error {
 	}
 
 	if !nodeConfig.NoFlannel {
+		if (nodeConfig.FlannelExternalIP) && (len(nodeConfig.AgentConfig.NodeExternalIPs) == 0) {
+			return fmt.Errorf("flannel-external-ip flag requires node-external-ip to be set")
+		} else if (nodeConfig.FlannelBackend != "wireguard-backend") && (nodeConfig.FlannelBackend != "ipsec") {
```

Comment:
Also, this might be confusing for agents... since --flannel-external-ip is set only on servers, but --node-external-ip is set on each individual node (including agents). Maybe just add a warning that this might not work right?

Reply:
Yes, better use the types in the daemons config package, thanks.

Like in the other PR, I was again wondering whether to fail the k3s process or just throw a warning. I chose fail because, even if agents don't see that config, things will not really work as intended. But I think both can work, so let me change to warning if you feel that might be better